### PR TITLE
a8n: Support GitHub CheckRuns

### PR DESCRIPTION
**Pull request by ryanslade**
Part of https://github.com/sourcegraph/sourcegraph/issues/7093

### CODE REVIEW

File: internal/extsvc/github/pulls.go [hunk 1]
```diff
@@ -53,13 +53,29 @@ type CheckSuite struct {
...
 	key := fmt.Sprintf("%s:%s:%s:%d", c.ID, c.Status, c.Conclusion, c.ReceivedAt.UnixNano())
 	return strconv.FormatUint(fnv1.HashString64(key), 16)
 }
 
+// CheckRun represents the status of a checkrun
+type CheckRun struct {
+	ID string
+	// One of COMPLETED, IN_PROGRESS, QUEUED, REQUESTED
+	Status string
+	// On of ACTION_REQUIRED, CANCELLED, FAILURE, NEUTRAL, SUCCESS, TIMED_OUT
```
Comment (unknwon) [inline internal/extsvc/github/pulls.go]: typo:
```suggestion
	// One of ACTION_REQUIRED, CANCELLED, FAILURE, NEUTRAL, SUCCESS, TIMED_OUT
```

Comment (unknwon) [reply internal/extsvc/github/pulls.go]: Thank god I found a typo, otherwise I feel so useless!

File: internal/extsvc/github/pulls.go [hunk 2]
```diff
@@ -53,13 +53,29 @@ type CheckSuite struct {
...
+	Conclusion string
```
Comment (unknwon) [inline internal/extsvc/github/pulls.go]: Could you add a docstring?

File: internal/extsvc/github/pulls.go (hunk 1 above)
Comment (ryanslade) [reply internal/extsvc/github/pulls.go]: hah, thanks :)

**Pull request merged**